hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -426,7 +426,13 @@ public boolean hasNext() {
 
     @Override
     public String next() {
-      String result = encodeFileUri(files.get(i), chksums != null? chksums.get(i) : null);
+      String result;
+      try {
+        result = ReplChangeManager.encodeFileUri(files.get(i), chksums != null ? chksums.get(i) : null, null);
+      } catch (IOException e) {
+        // If thrown IOException, it means, CM root is not set. So, just return original path.
+        result = files.get(i);
```

Comment:
i think .. ReplChangeManager should handle (null check is already there) ..cm_root not set scenario ..here in this layer we should throw exception for any failure

Reply:
This method is inherited and cannot change the definition. So, exception is caught and handle it. Yes, it is dead code for now.

Reply:
done